service/src/main/java/uk/gov/hmcts/reform/fpl/service/payment/FeeService.java

```diff
@@ -81,4 +104,11 @@ private FeeResponse makeRequest(FeeType feeType) throws FeeRegisterException {
             throw new FeeRegisterException(ex.status(), ex.getMessage(), ex);
         }
     }
+
+    private FeesData buildFeesDataFromFeeResponses(List<FeeResponse> feeResponses) {
+        return FeesData.builder()
+            .totalAmount(extractFeeToUse(feeResponses).map(FeeResponse::getAmount).get())
+            .fees(feeResponses.stream().map(FeeDto::fromFeeResponse).map(ElementUtils::element).collect(toList()))
```

Comment:
Can we consider making this a bit more readable somehow, quite long stream in single line a bit difficult to read.